docs/tuning_fledge.rst

```diff
@@ -762,6 +767,113 @@ When collection is enabled the following counters will be collected for the stor
+
+The frequency of running the purge process is very important, since it as the same effect as described for the purge system execution, but the impact is much higher. Consider a system that wants to retain data for 12 hours, if the purge process is set to run every 12 hours the number of readings over time would be as shown in the graph below
+
++---------------+
+| |PurgeCycles| |
++---------------+
+
+The red line indicates the configured retention point for the readings. Each point where the blue line drops is an execution of the purge process.
+
+This assumes we started with a system with no readings, we read in data for 12 hours and then run the purge process. This is shown as removing a small number of readings to reduce the retain readings to those less than 12 hours old. The initial run is in fact not likely to find any data to remove, or at most a handful of readings, depending on how long it takes the purge process to start executing.
+
+The system now continues to ingest data and will accumulate another 12 hours of data before purge is run again and the data reduced to the newest 12 hours of data.
+
+.. note::
+
+   We are assuming that either unsent data is not retained or we are sending all data north immediately it is received.
+
+This means that at a peak we are storing 24 hours of data, or twice what we wish to retain. Running the purge process more frequently than the retention period will not remove any more data than defined within the retention period, but will reduce the peaks of data that are stored. The other impact of this, not shown in the graph above, is that purge is **not an instantaneous process**. It takes time to purge the data and with some storage engines the system is blocked from ingesting more data during the purge. In this case the services will buffer the data in memory whilst waiting to gain access to the storage. Purging more often will decrease the number of readings that are removed for each execution and hence reduce the time that the ingest is locked out of the storage system. Reducing the time, and memory resources, that services have to buffer data in memory.
+
+.. note::
+
+   Since all data must go via the storage system to go from south service to the north services and tasks, the period when services are buffering in memory because the purge process is running, will increase the latency for data to traverse from the south to the north.
+
+There are many advantages to running the purge process more frequently than the retention period, however running it too frequently can cause increase in latency for readings and also if one purge process does not complete before another starts issues can be seen whereby the purge process dominates the usage of the storage subsystem and readings build up in the service memory buffers, eventually causing issues with excessive memory usage. The execution interval for the purge process must be balanced to not create issues with memory and storage utilisation.
```

Comment:
The first sentence is very long. May I suggest this?

There are many advantages to running the purge process more frequently than the retention period. Running it too frequently, however, can cause increase in latency for readings. In addition, if one purge process does not complete before another starts, issues can be seen whereby the purge process dominates the usage of the storage subsystem. If this happens, readings build up in the service memory buffers, eventually causing issues with excessive memory usage.